src/rgw/rgw_common.h

```diff
@@ -1839,3 +1839,32 @@ rgw_global_init(const std::map<std::string,std::string> *defaults,
 		    std::vector < const char* >& args,
 		    uint32_t module_type, code_environment_t code_env,
 		    int flags);
+
+class RgwAioCompletionWrapper {
```

Comment:
i really like the use of [RAII](https://en.cppreference.com/w/cpp/language/raii) to handle this cleanup 👍

but `RgwAioCompletionWrapper` is essentially just reinventing [std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr). if we add a custom deleter for `AioCompletion`, we can use `std::unique_ptr` directly:
```c++
// unique_ptr deleter that calls release()
struct AioCompletionDeleter {
  void operator()(librados::AioCompletion* c) { c->release(); }
};
using aio_completion_ptr = std::unique_ptr<librados::AioCompletion, AioCompletionDeleter>;
```
callers could either construct this ptr directly:
```c++
  aio_completion_ptr c{librados::Rados::aio_create_completion(nullptr, nullptr)};
```
or we could add a helper function that returns `aio_completion_ptr`:
```c++
  auto c = make_aio_completion(nullptr, nullptr);
```
what do you think?

Reply:
@cbodley could you please re-review?